### PR TITLE
patch output extension in csvtk/concat

### PR DESCRIPTION
**Pull request by rpetit3**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

@drpatelh haha as soon as you merged, I thought of something.

In the current implementation, you could theoretically get output files with random delimiter characters (e.g. if `out_format = '-'` the output file would be named `*.-` and thats an issue.

I think should make it either `tsv` or default to `csv`

### CODE REVIEW

**Comment by drpatelh** [comment]
Makes sense 👍🏽 Feel free to merge after you have fixed the tests.

File: modules/csvtk/concat/main.nf [hunk 1]
```diff
@@ -24,21 +24,22 @@ process CSVTK_CONCAT {
...
-    path "versions.yml"                     , emit: versions
+    tuple val(meta), path("${prefix}.{tsv,csv}"), emit: csv
+    path "versions.yml"                         , emit: versions
 
     script:
-    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
     def delimiter = in_format == "tsv" ? "\t" : (in_format == "csv" ? "," : in_format)
     def out_delimiter = out_format == "tsv" ? "\t" : (out_format == "csv" ? "," : out_format)
+    def out_extension = out_format == "tsv" ? 'tsv' : 'csv'
```
Comment (drpatelh) [inline modules/csvtk/concat/main.nf:34]:
```suggestion
    out_extension = out_format == "tsv" ? 'tsv' : 'csv'
```

File: modules/csvtk/concat/main.nf [hunk 2]
```diff
@@ -24,21 +24,22 @@ process CSVTK_CONCAT {
     val out_format
 
     output:
-    tuple val(meta), path("*.${out_format}"), emit: csv
-    path "versions.yml"                     , emit: versions
+    tuple val(meta), path("${prefix}.{tsv,csv}"), emit: csv
+    path "versions.yml"                         , emit: versions
```
Comment (drpatelh) [inline modules/csvtk/concat/main.nf:28]:
```suggestion
    tuple val(meta), path("*.${out_extension}"), emit: csv
    path "versions.yml"                       , emit: versions
```

Comment (rpetit3) [reply modules/csvtk/concat/main.nf]: I wonder if prefeix should be included to prevent capturing other CSVs/TSVs

```
cat 1.csv 2.csv 3.csv > 4.csv
```

Comment (drpatelh) [reply modules/csvtk/concat/main.nf]: Ah, ok. If more than one file has the same prefix then yes include it.

File: modules/csvtk/concat/main.nf [hunk 3]
```diff
@@ -24,21 +24,22 @@ process CSVTK_CONCAT {
     val out_format
 
     output:
-    tuple val(meta), path("*.${out_format}"), emit: csv
-    path "versions.yml"                     , emit: versions
+    tuple val(meta), path("${prefix}.${out_extension}"), emit: csv
+    path "versions.yml"                         , emit: versions
```
Comment (drpatelh) [inline modules/csvtk/concat/main.nf:28]:
```suggestion
    tuple val(meta), path("${prefix}.${out_extension}"), emit: csv
    path "versions.yml"                                , emit: versions
```

**Pull request merged**